### ISSUE TITLE
Unable to configure line length

### ISSUE THREAD

**Issue by Bennett-Lynch** [tnorbye/kdoc-formatter#53]
First of all, thanks for the great plugin! This is a huge time saver compared to manually formatting KDoc.

However, I am unable to successfully configure the line length when using the IntelliJ plugin. I have the following `.editorconfig` file in the root of my project:

```
root = true

[*.kt]
max_line_length=120
```

With this setting under "Code Style" enabled:

![image](https://user-images.githubusercontent.com/11811448/179849847-eaa9bcc2-d67f-4e99-a1a6-30dc3c67a63f.png)

But whenever I format, it uses the default line length of 72, which is too short for my preferences.

1. Am I doing anything wrong to configure the max line length?
2. Could a setting be added to the plugin preferences to specify the line length?

Version: 1.5.1

Thanks!

**Comment by tnorbye** [comment]
Try adding this to your .editorconfig:

```
[*.md]
max_line_length = 72
```

Note that kdoc-formatter has two separate concepts:
- the maximum  line length
- the maximum comment width

When you have deeply nested code, this distinction doesn't matter; the comments will break at the maximum line length. But for really wide code lines, the comments will not use the full line width. This is intentional, because for readability, you don't want really long lines -- those are harder to read (it's why books are typically in portrait orientation, not landscape). And for comments, readability is the primary concern, whereas for code, structure is the main concern.

I recently (in 1.5.1) changed the default from matching the line width to being 72. This is typically the default for markdown files, as well as git commit messages. Kdoc formatter will also initialize this value from the line-width configured for markdown files (hence the above [*.md] rule). Maybe this is a bit controversial, though I do think it makes for much nicer comments. At a minimum I should make sure it's easy to configure this for non-.editorconfig projects in IntelliJ.

**Comment by Bennett-Lynch** [comment]
Thank you for the reply @tnorbye.

I tried making your suggested change, but it's still not working (still defaulting to 72). I even tried changing the file mask to `[*]`, which also does not seem to be detected. Perhaps I'm doing something wrong?

In my project _root_, I have a file named `.editorconfig`, with the following contents:

```
root = true

[*]
max_line_length = 120
```

FWIW, I'm only trying to auto-format class-level and function-level KDoc. I'm not trying to format standalone markdown files (but I could see that being valuable as well).

I don't have a strong preference on the default behavior, as long as it's configurable, but I will add my $0.02: ~72 is a very practical width for commit messages, but I think the vast majority of codebases will consider it too short for function-level documentation.

**Comment by tnorbye** [comment]
I've attempted to fix this, in IDE plugin version 1.5.2. I haven't submitted it to JetBrains yet; can you try installing it to see if it works for you? 

With this there's a new setting at the bottom of the KDoc Formatting Options panel. 
<img width="777" alt="Screen Shot 2022-07-19 at 18 16 20" src="https://user-images.githubusercontent.com/223173/179875009-fc66d85d-3ec9-48db-b792-6cdb19dee7f2.png">

You can find the plugin zip to install (go to the Plugins settings page, click on the settings icon and then install plugin from disk). You can find the plugin zip on the 1.5.2 release page: https://github.com/tnorbye/kdoc-formatter/releases/tag/v1.5.2

**Comment by tnorbye** [comment]
I uploaded 1.5.2 to the plugin center and it's been approved and is public now.

**Comment by Bennett-Lynch** [comment]
Thanks @tnorbye for this quick fix! Just tested 1.5.2 and it works for my use case. Here's some feedback from testing it:

1. I'm a little bit confused by the wording "72 characters (or the configured Markdown line length)". If I configure my Markdown line length to be 120, it still appears to use 72:
![image](https://user-images.githubusercontent.com/11811448/180091774-9de7a49c-7fe9-4619-9e12-b0cf695535e1.png)
2. Unchecking the new option allows me to use my configured line length, BUT, if I change the line length, I have to restart the IDE for the plugin to recognize the new value.
3. This new option satisfies my personal needs out of this plugin, but I think it may still be more convenient to allow the line width to be specified directly in the plugin settings.

Thanks again!

**Comment by tnorbye** [comment]
Thanks for the note about the "collapse short comments" thing not working -- that was an embarassing copy paste error. It's been fixed in 1.5.5, just released. Also in 1.5.3 or 4 I added a new IDE setting which lets you manually configure override line lengths (and comment widths) in the IDE plugin which will override whatever the inferred .editorconfig settings are, which I think gives you what you need -- see the last two text fields here: https://github.com/tnorbye/kdoc-formatter/blob/main/screenshot-settings.png

(I've also done some preliminary work on block and line comments. From the command line you can format whole files with it; from the IDE only interactively the comment under the caret using the dedicated format kdoc action.)

If there are other bugs or remaining things I missed here, please file new separate issues. Thanks again.

**Comment by Bennett-Lynch** [comment]
Thanks @tnorbye! Love the built-in line width option.

FWIW, I'm still a little confused on "line width" vs "comment width". I thought the plugin only formats KDoc (comments)? Or is it supposed to refer to `//`-style comments?

Regardless, I'm happy with the current behavior, where the below configuration will correctly only auto-format KDoc to 100 width:

![image](https://user-images.githubusercontent.com/11811448/181655848-37d81757-15d2-4a8d-a6ac-59405ff4e515.png)

**Comment by Bennett-Lynch** [comment]
@tnorbye It looks like something broke between version 1.5.4 and 1.5.5 where the above config is no longer working. In 1.5.4 this would correctly format to 100 width. Now it's formatting to 72 again.

**Comment by tnorbye** [comment]
Ugh, I see it. Embarrassing. I really should try to write a UI test for this; the formatting machinery is well covered with unit tests but not the plugin UI code.  https://github.com/tnorbye/kdoc-formatter/pull/74

**Comment by tnorbye** [comment]
This is available in 1.5.6 now. I've uploaded it to plugins.jetbrains.com; approvals usually take around a day but you can install it manually from https://github.com/tnorbye/kdoc-formatter/releases/tag/v1.5.6
